### ISSUE TITLE
Support for non-default HTTP ports

### ISSUE THREAD

**Issue by solarispika** [sindresorhus/is-reachable#55]
Hello,

I have some HTTP servers running on ports other than 80 and 443, which are hosted by docker containers.
Those containers are run on demand by users, and I would like to wait until they can handle HTTP requests before returning to user.
But sometimes the module fails to wait until server is ready.

I found in the module that ports other than 80 and 443 are checked simply by port existence, but not HTTP response.
I think it is "docker opening up the port before running the HTTP server" that cause the problem.

After a simple experiment of forced HTTP checking on non-standard ports, it works as expected.
Is it possible to support such scenario?

Thanks.

**Comment by sindresorhus** [comment]
Yeah. Maybe we should change the HTTP check to run as long as the scheme is `http` or `https` no matter the port?

**Comment by solarispika** [comment]
Yes, your proposal looks feasible and simpler than what I did.
What I originally tried is similar to https://github.com/sindresorhus/is-reachable/pull/42

Also, I found that assigning default HTTP(s) port to an corresponding URL object without port does not do anything useful, as the port is still empty in the result URL object. (tested under node.js v16)
```javascript
        if (!url.port) {
                url.port = url.protocol === 'http:' ? 80 : 443;
        }
```
I think that it can be simplified to just assign the default port 443 when port is not given and protocol is neither `http` nor `https`.

If there is no other consideration, I can make a PR for you later.
